com.unity.render-pipelines.universal/Runtime/ForwardRenderer.cs

```diff
@@ -563,5 +578,50 @@ bool CanCopyDepth(ref CameraData cameraData)
             bool msaaDepthResolve = false;
             return supportsDepthCopy || msaaDepthResolve;
         }
+
+        // (ASG)
+        /// <summary>
+        /// Some effects like tonemap and color grading can be applied in the ForwardPass without needing a separate
+        /// post process shader. This function returns whether all active post-processing effects are compatible to be
+        /// run in the ForwardPass.
+        /// </summary>
+        /// <remarks>Expensive, because we re-query the effects stack. Don't run more than once.</remarks>
+        public bool CanSkipSeparatePostProcessPass()
```

Comment:
Think we could calculate this in `VolumeStack.Reload()` and cache it in a field?

Reply:
We could. It's not terribly slow, just not something to run in a tight loop. This is very similar to the code in ColorGradingLutPass, so it should be fast enough to run every frame. 

I'm actively trying to avoid touching very much in these files, so I'd prefer this simpler route, unless it proves to be a performance issue. VolumeStack also technically shouldn't know much about the URP ForwardRenderer, since it's used across URP and HDRP, etc.